### PR TITLE
Implement Websocket server compression

### PR DESCRIPTION
**Pull request by phanidevavarapu**
Resolves https://github.com/open-telemetry/opamp-go/issues/121

### CODE REVIEW

File: server/serverimpl_test.go [hunk 1]
```diff
@@ -224,6 +229,103 @@ func TestServerReceiveSendMessage(t *testing.T) {
...
+			settings := &StartSettings{Settings: Settings{Callbacks: callbacks, EnableCompression: withCompression}}
+			srv := startServer(t, settings)
+			defer srv.Stop(context.Background())
+
+			// We use a transparent TCP proxy to be able to count the actual bytes transferred so that
+			// we can test the number of actual bytes vs number of expected bytes with and without compression.
+			proxy := testhelpers.NewProxy(settings.ListenEndpoint)
+			assert.NoError(t, proxy.Start())
+
+			clientSettings := *settings
```
Comment (tigrannajaryan) [inline server/serverimpl_test.go:273]:
```suggestion
			serverSettings := *settings
```

**Review by tigrannajaryan — COMMENTED** [review]
Thank you for the PR @phanidevavarapu 
Looks mostly good to me, just one small comment.

File: server/serverimpl_test.go (hunk 1 above)
Comment (phanidevavarapu) [reply server/serverimpl_test.go]: Done

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@phanidevavarapu merged, thank you!

**Pull request merged**